src/Executables/DumpEquationOfStateForRotNS/DumpEquationOfStateForRotNS.cpp

```diff
@@ -0,0 +1,181 @@
+#include <string>
+#include <utility>
+#include <vector>
+
+#include "DataStructures/DataVector.hpp"
+#include "Options/Options.hpp"
+#include "Options/ParseOptions.hpp"
+#include "Parallel/Printf.hpp"
+#include "PointwiseFunctions/Hydro/EquationsOfState/EquationOfState.hpp"
+#include "PointwiseFunctions/Hydro/Tags.hpp"
+#include "PointwiseFunctions/Hydro/Units.hpp"
+#include "Utilities/ErrorHandling/Error.hpp"
+#include "Utilities/FileSystem.hpp"
+#include "Utilities/TMPL.hpp"
+
+// Charm looks for this function but since we build without a main function or
+// main module we just have it be empty
+extern "C" void CkRegisterMainModule(void) {}
+
+namespace {
+void dump_barotropic_eos(
+    const EquationsOfState::EquationOfState<true, 1>& eos,
+    const size_t number_of_log10_number_density_points_for_dump,
+    const std::string& output_file_name,
+    const double lower_bound_rest_mass_density_cgs,
+    const double upper_bound_rest_mass_density_cgs) {
+  using std::log10;
+  using std::pow;
+  const double log10_lower_bound_number_density_cgs =
+      log10(lower_bound_rest_mass_density_cgs / hydro::units::cgs::baryon_mass);
+  const double log10_upper_bound_number_density_cgs =
+      log10(upper_bound_rest_mass_density_cgs / hydro::units::cgs::baryon_mass);
+  const double delta_log_number_density_cgs =
+      (log10_upper_bound_number_density_cgs -
+       log10_lower_bound_number_density_cgs) /
+      static_cast<double>(number_of_log10_number_density_points_for_dump - 1);
+
+  if (file_system::check_if_file_exists(output_file_name)) {
+    ERROR_NO_TRACE("File " << output_file_name
+                           << " already exists. Refusing to overwrite.");
+  }
+  std::ofstream outfile(output_file_name.c_str());
+
+  for (size_t log10_number_density_index = 0;
+       log10_number_density_index <
+       number_of_log10_number_density_points_for_dump;
+       ++log10_number_density_index) {
+    using std::pow;
+    const double number_density_cgs =
+        pow(10.0, log10_lower_bound_number_density_cgs +
+                      static_cast<double>(log10_number_density_index) *
+                          delta_log_number_density_cgs);
+    const double rest_mass_density_cgs =
+        number_density_cgs * hydro::units::cgs::baryon_mass;
+
+    const Scalar<double> rest_mass_density_geometric{
+        rest_mass_density_cgs / hydro::units::cgs::rest_mass_density_unit};
+    const Scalar<double> pressure_geometric =
+        eos.pressure_from_density(rest_mass_density_geometric);
+    const Scalar<double> specific_internal_energy_geometric =
+        eos.specific_internal_energy_from_density(rest_mass_density_geometric);
+    const Scalar<double> total_energy_density_geometric{
+        get(rest_mass_density_geometric) *
+        (1.0 + get(specific_internal_energy_geometric))};
+
+    const double total_energy_density_cgs =
+        get(total_energy_density_geometric) *
+        hydro::units::cgs::rest_mass_density_unit;
```

Comment:
Maybe add a comment that the table stores energy_density/c^2, since for energy_density in cgs you would need to multiply by that ( i.e. rest-mass part is rho c^2)